### PR TITLE
Rename `parseFieldValueComponents` to `parseMultipleHeaderFields`

### PR DESCRIPTION
**Pull request by rekryt**
compatibility to http v2.0.0

### CODE REVIEW

**Comment by rekryt** [comment]
Tests fixed

File: src/Rfc6455Connector.php [hunk 1]
```diff
@@ -111,7 +111,7 @@ private function generateRequest(WebsocketHandshake $handshake, string $key): Re
         $request->setTlsHandshakeTimeout($handshake->getTlsHandshakeTimeout());
         $request->setHeaderSizeLimit($handshake->getHeaderSizeLimit());
 
-        $extensions = \array_column(Http\parseFieldValueComponents($request, 'sec-websocket-extensions') ?? [], 0, 0);
+        $extensions = \array_column(Http\parseMultipleHeaderFields($request, 'sec-websocket-extensions') ?? [], 0, 0);
```
Comment (kelunik) [inline src/Rfc6455Connector.php]: We can use parseHeaderTokens here.

Comment (rekryt) [reply src/Rfc6455Connector.php]: ok, updated

**Comment by rekryt** [comment]
Need tag of amphp/http-server
v3.0.0-beta.7 (^3) is less relevant than 3.x-dev

File: test/WebsocketConnectionTest.php [hunk 2]
```diff
@@ -35,7 +42,16 @@ protected function createServer(WebsocketClientHandler $clientHandler): array
...
+        $serverSocketFactory = new ConnectionLimitingServerSocketFactory(
+            new LocalSemaphore(self::DEFAULT_CONNECTION_LIMIT)
+        );
+        $clientFactory = new ConnectionLimitingClientFactory(
+            new SocketClientFactory($logger),
+            $logger,
+            self::DEFAULT_CONNECTIONS_PER_IP_LIMIT
+        );
+
+        $httpServer = new SocketHttpServer($logger, $serverSocketFactory, $clientFactory);
```
Comment (trowski) [inline test/WebsocketConnectionTest.php:54]: You can use `SocketHttpServer::createForDirectAccess()` instead.

File: src/Rfc6455Connector.php [hunk 3]
```diff
@@ -71,7 +71,7 @@ public function connect(WebsocketHandshake $handshake, ?Cancellation $cancellati
                 return;
             }
 
-            $extensions = \array_column(Http\parseMultipleHeaderFields($response, 'sec-websocket-extensions') ?? [], 0, 0);
+            $extensions = \array_column(Http\parseHeaderTokens($response, 'sec-websocket-extensions') ?? [], 0, 0);
```
Comment (kelunik) [inline src/Rfc6455Connector.php:74]: array_column can and should be dropped then, no?

**Comment by bennnjamin** [comment]
I need to get this merged so that I can update all http-* packages in my project to the latest version. Currently, this pull request is blocking any further updates to the http-* packages since this package also relies on http v2. What can I do to help move this along? Happy to open another PR if needed.

**Comment by trowski** [comment]
Resolved by https://github.com/amphp/websocket-client/commit/24daf3cb8cbf25aa3d16a9434ca72e0af9b005b5 and tagged in [2.0.0 Beta 3](https://github.com/amphp/websocket-client/releases/tag/v2.0.0-beta.3).

**Pull request closed without merging**